src/Build/BuildCheck/API/ConfigurationContext.cs

```diff
@@ -20,19 +20,9 @@ private ConfigurationContext(CustomConfigurationData[] customConfigurationData)
 
     internal static ConfigurationContext FromDataEnumeration(CustomConfigurationData[] customConfigurationData)
     {
-        if (!customConfigurationData.Any(BuildCheck.CustomConfigurationData.NotNull))
```

Comment:
@JanKrivanek could you please let me know if there was a use case to filter out null values from the configurationContext of custom configs? Having that, the index of the retrieved custom config will depend on the fact that the custom data retrieved from the editorconfig. Am I missing something?

Reply:
The consuming BuildCheck should not rely on indexes, but rather on the RuleId that's part of the ConfigurationData.

It is very likely that wast majority of the BuildChecks/Rules won't have any CustomData at all. So the original idea was - under the assumption that we'd be passing only the CustomData - to optimize that case by passing single empty object.

With the recent finding that we'll need to pass the Infra configuration as well - this optimization doesn't make that much sense anymore

Reply:
Thank you for the clarifications!